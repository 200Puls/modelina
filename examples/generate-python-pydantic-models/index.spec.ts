const spy = jest.spyOn(global.console, 'log').mockImplementation(() => { return; });
import {generate} from './index';

describe('Should be able to render python models', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    //Generate is called 2x, so even though we expect 2 models, we double it
    expect(spy.mock.calls.length).toEqual(4);
    expect(spy.mock.calls[0]).toMatchSnapshot('class-model');
    expect(spy.mock.calls[1]).toMatchSnapshot('enum-model');
  });
});
