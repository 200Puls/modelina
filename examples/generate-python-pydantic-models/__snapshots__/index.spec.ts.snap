// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to render python models and should log expected output to console 1`] = `
Array [
  "from typing import Optional, Any
from pydantic import BaseModel, Field
class Root(BaseModel): 
  optionalField: Optional[str] = Field(alias='this field is optional')
  noDescription: str = Field()"
]
`;
